intellij/src/saros/intellij/ui/views/buttons/AddContactButton.java

```diff
@@ -0,0 +1,167 @@
+
+    SarosPluginContext.initComponent(this);
+
+    connectionHandler.addConnectionStateListener(connectionStateListener);
+
+    addActionListener(actionEvent -> showAddContactDialog());
+
+    if (!connectionHandler.isConnected()) {
+      setEnabled(false);
+    }
+  }
+
+  @Override
+  public void dispose() {
+    connectionHandler.removeConnectionStateListener(connectionStateListener);
+  }
+
+  /**
+   * Shows the dialogs requesting the JID and (optional) nickname for the account to add as a
+   * contact.
+   *
+   * <p>Calls {@link XMPPContactsService#addContact(JID, String, BiPredicate)} if the dialogs were
+   * not canceled.
+   */
+  private void showAddContactDialog() {
+
+    // TODO better communicate issue to user
+    // TODO use checkInput as it disables the OK button instead of just doing nothing when OK is
+    //  clicked on illegal input.
+    InputValidator inputValidator =
+        new InputValidator() {
+          @Override
+          public boolean checkInput(String inputString) {
+            return true;
+          }
+
+          @Override
+          public boolean canClose(String inputString) {
+            JID jid = new JID(inputString);
+
+            return jid.isValid() && !jid.getName().isEmpty();
+          }
+        };
+
+    String userID;
+
+    try {
+      userID =
+          SafeDialogUtils.showInputDialog(
+              project,
+              Messages.AddContactButton_contact_jid_dialog_message,
+              Messages.AddContactButton_contact_jid_dialog_initial_input,
+              Messages.AddContactButton_contact_jid_dialog_title,
+              inputValidator,
+              new TextRange(0, 0));
+
+    } catch (IllegalAWTContextException e) {
+      log.error("Account creation failed.", e);
+
+      // TODO show error?
+      // showAccountCreationFailedError(e);
+
+      return;
+    }
+
+    if (userID == null) {
+      log.debug("Adding contact canceled by user during user id entry.");
+
+      return;
+    }
+
+    JID jid = new JID(userID);
+
+    String nickname;
+
+    try {
+      nickname =
+          SafeDialogUtils.showInputDialog(
+              project,
+              Messages.AddContactButton_contact_nickname_dialog_message,
+              "",
+              Messages.AddContactButton_contact_nickname_dialog_title);
+
+    } catch (IllegalAWTContextException e) {
+      log.error("Account creation failed.", e);
+
+      // TODO show error?
+      // showAccountCreationFailedError(e);
+
+      return;
+    }
+
+    if (nickname == null) {
+      log.debug("Adding contact canceled by user during user nickname entry.");
+
+      return;
+    }
+
+    BiPredicate<String, String> dialog =
+        (title, message) -> {
+          log.fatal("showing dialog " + title + " - " + message);
+
+          Integer integer;
+
+          try {
+            integer = SafeDialogUtils.showYesNoDialog(project, message, title);
```

Comment:
I do not know if it is a good idea to return null as an integer here ? And when is null returned ?

Reply:
It can't return null (unless the dialog aborts due to an internal error?). The usage of `Integer` stems from the dialog being executed synchronously in a runnable. [Here is the code.](https://github.com/saros-project/saros/blob/9b13dea8df827a2aea8d7b2b52b1c763d6f6040b/intellij/src/saros/intellij/ui/util/SafeDialogUtils.java#L203) To fetch the result, an `AtomicReference<>` is used, which requires an object type. But keeping it as an Integer for the return value is unnecessary.

That is an oversight on my part. It should unpack the int (or, even better, do the check against `Messages#YES` and `Messages#No` and return a `boolean`). Will adjust in a separate patch.